### PR TITLE
Release 2.5.0

### PR DESCRIPTION
**Pull request by delano**
Collects changelog fragments into CHANGELOG.rst via scriv, bumps VERSION to 2.5.0, and promotes `json_schemer` from a development dependency to a runtime dependency.

Changes shipping in this release: re_encrypt_fields! fix (#235), RecordExistsError#existing_id (#242), destroy! index cleanup (#241), unique_index split-identifier tryout (#243), atomic_swap race fix (#247).

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -7,6 +7,92 @@ The format is based on `Keep a Changelog <https://keepachangelog.com/en/1.1.0/>`
...
+AI Assistance
+-------------
+
+- Collaborated with Claude on isolating the no-op root cause (the setter's ConcealedString-preservation branch), drafting the raw-envelope regression canary that inspects ``key_version`` in stored JSON, reworking ``examples/encrypted_fields.rb`` to exercise the real rotation flow rather than pre-assigning plaintext, and adding edge-case coverage for nonce freshness, missing-old-key failures, type-guard assertions, and mixed encrypted/plain/transient field models. Issue #235
+
+- Implementation and test coverage drafted with Claude Code
+  (backend-dev + qa-automation-engineer subagents), reviewed by
+  feature-dev:code-reviewer.
+
+- Claude Code (Opus 4.7) drafted the new ``unique_index_split_identifier_try.rb``
```
[the remark this answers is not in the document]
Comment (delano) [reply CHANGELOG.rst:82]: 🐾
